### PR TITLE
Improve MessagePack model

### PR DESCRIPTION
**Pull request by jarmuszz**
Current implementation of the MessagePack model allows for creation of malformed objects as the `ByteVector` class (which underlays most of the types in the model) has a maximum length of `Long.MaxValue` bytes which is bigger than the [specification](https://github.com/msgpack/msgpack/blob/master/spec.md) defines for some types. 

This PR proposes following changes:
- `Float32(ByteVector)` -> `Float32(Float)`
- `Float64(ByteVector)` -> `Float64(Double)`
- `Timestamp32(ByteVector)` -> `Timestamp64(Int)`
- `Timestamp64(ByteVector, ByteVector)` -> `Timestamp64(Long)` with `nanoseconds` and `seconds` fields
- `Timestamp96(ByteVector, ByteVector)` -> `Timestamp96(Int, Long)` 

Ref. #603

### CODE REVIEW

**Comment by jarmuszz** [comment]
I have currently held back from changing the integer types. `UnsignedInt` would have to hold a `BigInt` in order to keep a sign which is not quite performant. We could also bit cast it into a `Long` but I think it would be confusing for users.

File: msgpack/src/main/scala/fs2/data/msgpack/low/model.scala [hunk 1]
```diff
@@ -40,9 +40,18 @@ object MsgpackItem {
...
-  case class Timestamp96(nanoseconds: ByteVector, seconds: ByteVector) extends MsgpackItem
+  case class Timestamp32(seconds: Int) extends MsgpackItem
+
+  /** Stores data in a 30-bit [[nanoseconds]] and a 34-bit [[seconds]] fields, both of which are accessible as class
+    * attributes. To ensure valid data length at the type level, both fields are constructed from a single 64-bit
+    * [[combined]] variable.
+    * @param combined [[nanoseconds]] and [[seconds]] combined into a signle 64-bit value
+    */
+  case class Timestamp64(combined: Long) extends MsgpackItem {
+    val nanoseconds: Int = (combined >> 34).toInt // we are sure that (x: Long) >> 34 fits in an int
```
Comment (satabin) [inline msgpack/src/main/scala/fs2/data/msgpack/low/model.scala:51]: When using the arithmetic right shift, it is better to apply a mask before converting.

```suggestion
    val nanoseconds: Int = (0xffffffffL & (combined >> 34)).toInt // we are sure that (x: Long) >> 34 fits in an int
```

Comment (satabin) [reply msgpack/src/main/scala/fs2/data/msgpack/low/model.scala:51]: (note that this mask has only 7 f's and not 8 like the other one)

Comment (jarmuszz) [reply msgpack/src/main/scala/fs2/data/msgpack/low/model.scala:51]: I guess without the mask we could end up padding with 1s if the number is negative? Is that right?

Comment (satabin) [reply msgpack/src/main/scala/fs2/data/msgpack/low/model.scala:51]: Yes, this is what can happen.

**Review by satabin — APPROVED** [review]
Looks good to me, let's see if @ybasket has some comments.

**Comment by ybasket** [comment]
> Looks good to me, let's see if @ybasket has some comments.

No, was only waiting for the small fix you had requested. LGTM

**Pull request merged**